### PR TITLE
Disallow types named 'scoped'

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/62931

The breaking change was documented in https://github.com/dotnet/roslyn/pull/62788/files#diff-b58628cb08a707ff68b2546a1ad0ddf3a900aebb0ba899063c3ffe0a8ae7f07d

Relates to test plan https://github.com/dotnet/roslyn/issues/59194

### CODE REVIEW

**Comment by alrz** [comment]
All lowercase types are already disallowed, no?

**Comment by jcouv** [comment]
> All lowercase types are already disallowed, no?

We only produce a warning wave warning for those. So we escalate to an error when we actually start using a lowercase word as contextual keyword.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs [hunk 1]
```diff
@@ -478,7 +478,8 @@ internal static void ReportReservedTypeName(string? name, CSharpCompilation comp
 
             if (reportIfContextual(SyntaxKind.RecordKeyword, MessageID.IDS_FeatureRecords, ErrorCode.WRN_RecordNamedDisallowed)
```
Comment (Youssef1313) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs:479]: Curious why records is warning and other contextual keywords are errors?

Comment (jcouv) [reply src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs:479]: Would have to dig up the PR and/or LDM notes. Most likely this was evolution of our philosophy towards such breaks. Records came before we added the warning wave warning.

**Review by 333fred — APPROVED** [review]

**Review by cston — APPROVED** [review]

**Pull request merged**